core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -803,25 +807,10 @@ public Tenant getTenant(String tenantUniqueID) throws UserStoreException {
                 tenant.setRealmConfig(realmConfig);
                 setSecondaryUserStoreConfig(realmConfig, id);
                 tenant.setAdminName(realmConfig.getAdminUserName());
-                if (StringUtils.isNotBlank(tenant.getAssociatedOrganizationUUID())) {
-                    String adminId = realmConfig.getAdminUserId();
-                    if (StringUtils.isNotBlank(adminId)) {
-                        tenant.setAdminUserId(adminId);
-                    } else {
-                        // If realms were not migrated after https://github.com/wso2/product-is/issues/14001.
-                        try {
-                            /*
-                            Try to resolve user id from username. This get successful if the user store manager
-                            can resolve the user id even when the user does not belong to the tenant user store
-                             */
-                            tenant.setAdminUserId(getUserId(realmConfig.getAdminUserName(), id));
-                        } catch (UserStoreException ex) {
-                            // Failure to resolve user id from username means the user id is stored as the username.
-                            tenant.setAdminUserId(realmConfig.getAdminUserName());
-                        }
-                    }
+                if (tenant.getAssociatedOrganizationUUID() != null) {
+                    tenant.setAdminUserId(resolveTenantAdminUserId(realmConfig, tenant));
                 } else {
-                    tenant.setAdminUserId(getUserId(realmConfig.getAdminUserName(), id));
+                    tenant.setAdminUserId(getUserId(tenant.getAdminName(), tenant.getId()));
```

Comment:
As the tenant object has set the values, using those values would make sense the parameters passed to the `getUserI` method